python/services/dataservice/dmod/test/it_data_derive_util.py

```diff
@@ -0,0 +1,199 @@
+import unittest
+import os
+from datetime import datetime
+from pathlib import Path
+from dotenv import load_dotenv
+from dmod.core.meta_data import DataCategory
+from dmod.modeldata.data.object_store_manager import ObjectStoreDatasetManager
+from dmod.dataservice.dataset_manager_collection import DatasetManagerCollection
+from ..dataservice.data_derive_util import DataDeriveUtil
+from typing import List, Optional
+
+
+class TestBase:
+
+    _TEST_ENV_FILE_BASENAME = ".test_env"
+
+    @classmethod
+    def find_project_root(cls, path: Optional[Path] = None) -> Path:
+        if path is None:
+            path = Path('.').resolve()
+        original = path
+        while True:
+            if path.is_dir() and path.joinpath("example.env").exists() and path.joinpath("docker").exists():
+                return path
+            elif path == path.anchor:
+                break
+            else:
+                path = path.parent
+        raise RuntimeError(f"Can't identify project root starting from {original!s}")
+
+    @classmethod
+    def source_test_env_file(cls):
+        # Find the global .test_env file from project root, and source
+        proj_root = cls.find_project_root()
+        test_env = proj_root.joinpath(cls._TEST_ENV_FILE_BASENAME)
+
+        if test_env.exists():
+            load_dotenv(dotenv_path=str(test_env))
+
+
+try:
+    import pytest
+    TestBase.source_test_env_file()
+    explicit_on = os.environ.get("DERIVE_UTIL_IT_ON", "false").strip().lower() == "true"
+    skip_unless_explicit = pytest.mark.skipif(not explicit_on,
+                                              reason=f"IntegrationTestDataDeriveUtil tests skipped locally; you can "
+                                                     f"activate by setting 'DERIVE_UTIL_IT_ON' to 'true' in your "
+                                                     f"project '{TestBase._TEST_ENV_FILE_BASENAME}' file.")
+except ImportError as e:
+    raise RuntimeError(f"Failed to execute tests due to error with check of whether special skip logic applies") from e
+
+
+@skip_unless_explicit
```

Comment:
If `pytest` is installed by you run this with `unittest` what happens? I would think that this is a no-op?

Reply:
I've made a modification to how this is handled that should handle that situation more appropriate.